libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp

```diff
@@ -498,14 +505,27 @@ void AP_InertialSensor_Backend::update_gyro(uint8_t instance)
         _last_gyro_filter_hz[instance] = _gyro_filter_cutoff();
     }
     // possily update the notch filter parameters
-    if (_last_notch_center_freq_hz[instance] != _gyro_notch_center_freq_hz() ||
-        _last_notch_bandwidth_hz[instance] != _gyro_notch_bandwidth_hz() ||
+    if (!is_equal(_last_notch_center_freq_hz[instance], _gyro_notch_center_freq_hz()) ||
+        !is_equal(_last_notch_bandwidth_hz[instance], _gyro_notch_bandwidth_hz()) ||
         !is_equal(_last_notch_attenuation_dB[instance], _gyro_notch_attenuation_dB())) {
         _imu._gyro_notch_filter[instance].init(_gyro_raw_sample_rate(instance), _gyro_notch_center_freq_hz(), _gyro_notch_bandwidth_hz(), _gyro_notch_attenuation_dB());
         _last_notch_center_freq_hz[instance] = _gyro_notch_center_freq_hz();
         _last_notch_bandwidth_hz[instance] = _gyro_notch_bandwidth_hz();
         _last_notch_attenuation_dB[instance] = _gyro_notch_attenuation_dB();
     }
+    // possily update the harmonic notch filter parameters
```

Comment:
To follow the code in the _notify method, I'd prefer to see the code below be before the existing notch code.

Reply:
ok